src/compiler/utilities.ts

```diff
@@ -5012,7 +5012,7 @@ namespace ts {
             || kind === SyntaxKind.BooleanKeyword
             || kind === SyntaxKind.StringKeyword
             || kind === SyntaxKind.SymbolKeyword
-            || kind === SyntaxKind.ThisKeyword
+            || kind === SyntaxKind.ThisType

```

Comment:
`ThisType` is already within the range of `FirstTypeNode` through `LastTypeNode`.